.github/workflows/kotlin.yml

```diff
@@ -0,0 +1,66 @@
+name: Kotlin
+on:
+  pull_request:
+    paths:
+      - "kotlin/**"
+      - ".github/workflows/kotlin.yml"
+  merge_group:
+    types: [checks_requested]
+  workflow_call:
+
+# Cancel old workflow runs if new code is pushed
+concurrency:
+  group: "kotlin-${{ github.workflow }}-${{ github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  draft-release:
```

Comment:
I don't think that we should draft releases in a language-specific CI steps, maybe only in the `cd` workflow?

Reply:
Hm, yeah we'll need to think about how we plan to ship releases in the monorepo structure. For now, we could release all components in unison until we have a better strategy.

The issue here is that we depend on this step to compute a `version` that will be used during the client app build, and used as a key to upload the artifact to the drafted release for manual QA if needed.

After QA is complete, "publishing" a release would publish any downloadable artifacts that were previously attached in the `draft release` step, and also build+publish the client applications to their respective stores for distribution to closed beta users.

This is just off the top of my head, perhaps @pratikvelani or @roop has input here.

Reply:
The `draft-release` step is _supposed_ to re-use the existing previously drafted release if it exists so hopefully we shouldn't have unbounded numbers of drafted releases, but if that turns out not to be the case, let's find another solution